libs/core/tests/unit_tests/output_parsers/test_json.py

```diff
@@ -235,10 +235,12 @@ def test_parse_json_with_python_dict() -> None:
     ('{"foo": "bar", "bar": "foo', '{"foo": "bar", "bar": "foo"}'),
     ('{"foo": "bar", "bar": "foo}', '{"foo": "bar", "bar": "foo}"}'),
```

Comment:
The original test was correct, the updated test modifies the output incorrectly.

You can test it with this version which is a bit easier to understand:

```
def test_streaming_variant() -> None:
    obj = {
        "a": "b\\",
        "c": "d",
    }

    parts = json.dumps(obj)
    sub_strings = [parts[:idx] for idx in range(1, len(parts))]
    results = [parse_partial_json(sub_string) for sub_string in sub_strings]

    assert results == [
        {},
        {},
        {},
        {},
        {},
        {},
        {"a": ""},
        {"a": "b"},
        {"a": "b\\"},
        # Output below is wrong, there should only be a single `\`
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\"},
        {"a": "b\\\\", "c": ""},
        {"a": "b\\\\", "c": "d"},
        {"a": "b\\\\", "c": "d"},
    ]
```